be/src/exprs/json_functions.cpp

```diff
@@ -157,11 +157,34 @@ Status JsonFunctions::extract_from_object(simdjson::ondemand::object& obj, const
         // we have to do some special treatment for the second elem of json path.
         // If the key is not found in json object, simdjson::NO_SUCH_FIELD would be returned.
         if (i == 1) {
-            HANDLE_SIMDJSON_ERROR(obj.find_field_unordered(col).get(tvalue),
-                                  fmt::format("unable to find field: {}", col));
+            if (col == "*") {
+                // There should be no jsonpath for this pattern, $.*
+                return Status::InvalidArgument(fmt::format("invalid json path: {}", jsonpath[i].key));
+            } else {
+                HANDLE_SIMDJSON_ERROR(obj.find_field_unordered(col).get(tvalue),
+                                      fmt::format("unable to find field: {}", col));
+            }
         } else {
-            HANDLE_SIMDJSON_ERROR(tvalue.find_field_unordered(col).get(tvalue),
-                                  fmt::format("unable to find field: {}", col));
+            if (col == "*") {
+                // There are always two patterns.
+                // 1. {"field_name": null}
+                // 2. {"field_name": {"field_type": data}}
+                // For pattern1, we just return null value.
+                // For pattern2, we get the first field of object as next value.
+                if (tvalue.is_null()) {
+                    return Status::NotFound("null value");
+                }
+                if (tvalue.type() != simdjson::ondemand::json_type::object) {
+                    return Status::InvalidArgument(fmt::format("invalid json path: {}", jsonpath[i].key));
```

Comment:
I think it's better to report the error as `extracting * from non-object type is not supported`, or something else.